src/kubesystem/kubesystem.go

```diff
@@ -20,6 +23,23 @@ func GetUID(clt client.Reader) (types.UID, error) {
 	return kubeSystemNamespace.UID, nil
 }
 
-func DeployedViaOLM() bool {
-	return os.Getenv("DEPLOYED_VIA_OLM") == "true"
+func IsDeployedViaOlm(clt client.Reader, podName string, podNamespace string) (bool, error) {
+	pod := &corev1.Pod{}
+	err := clt.Get(context.TODO(), types.NamespacedName{Name: podName, Namespace: podNamespace}, pod)
+	if err != nil {
+		return false, err
+	}
+	if _, ok := pod.Annotations[olmSpecificAnnotation]; ok {
+		return true, nil
+	} else {
+		return false, nil
+	}
+}
+
+func CreateDefaultClient() (client.Client, error) {
+	kubeCfg, err := config.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+	return client.New(kubeCfg, client.Options{})
```

Comment:
error is external, needs to be wrapped
```suggestion
    kubeClient, err := client.New(kubeCfg, client.Options{})
	return kubeClient, errors.WithStack(err)
```